service/service.go

```diff
@@ -0,0 +1,181 @@
+package service
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/signal"
+
+	"github.com/ONSdigital/dp-dimension-extractor/dimension"
+	"github.com/ONSdigital/dp-dimension-extractor/schema"
+	"github.com/ONSdigital/go-ns/avro"
+	"github.com/ONSdigital/go-ns/kafka"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/s3"
+)
+
+// ----------------------------------------------------------------------------
+
+// Service represents the necessary config for dp-dimension-extractor
+type Service struct {
+	EnvMax       int64
+	Consumer     *kafka.ConsumerGroup
+	ImportAPIURL string
+	Producer     kafka.Producer
+	S3           *s3.S3
+}
+
+type inputFileAvailable struct {
+	FileURL    string `avro:"file_url"`
+	InstanceID string `avro:"instance_id"`
+}
+
+type dimensionExtracted struct {
+	FileURL    string `avro:"file_url"`
+	InstanceID string `avro:"instance_id"`
+}
+
+// ----------------------------------------------------------------------------
+
+// Start handles consumption of events ad produces new events
+func (svc *Service) Start() {
+	log.Info("application started", log.Data{"import-api-url": svc.ImportAPIURL})
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, os.Kill)
+
+	healthChannel := make(chan bool)
```

Comment:
channel is a variable name is probably not necessary - if it's not clear just as health, healthc or healthCh are standard options